source/common/grpc/stat_names.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "envoy/grpc/status.h"
+
+#include "common/stats/symbol_table_impl.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Grpc {
+
+/**
+ * Captures symbolized representation for tokens used in grpc stats. These are
+ * broken out so they can be allocated early and used across all gRPC-related
+ * filters.
+ */
+struct StatNames {
```

Comment:
do you think it would make sense to provide a base class for these stat name helpers that just has the ctor + the pool field? not sure if it's worth to try to DRY it up given how little code it already is (and it can certainly be addressed in another PR)

Reply:
I don't think it would save anything. You'd have to have a constructor for the derived class to pass the SymbolTable to the base-class constructor; you might well just init the member var.

I guess it saves the member-var declaration but replaces it with an inheritance declaration so it might be one less line of code.